From 9e51f938dc99c049da59f60bc588b229b04a2655 Mon Sep 17 00:00:00 2001
From: Silvio Traversaro <silvio@traversaro.it>
Date: Fri, 14 Apr 2023 15:17:08 +0200
Subject: [PATCH 2/4] avoid conflicting with onnxruntime.dll in system32

apparently Windows ships a onnxruntime.dll file in C:\Windows\system32\onnxruntime.dll, and according to .dll
search order in Windows (https://docs.microsoft.com/en-us/windows/win32/dlls/dynamic-link-library-search-order),
anything that is found in C:\Windows\system32 has a higher priority w.r.t. to anything found in the PATH.
See microsoft/onnxruntime#11230 for more details.
---
 cmake/onnxruntime.cmake | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/cmake/onnxruntime.cmake b/cmake/onnxruntime.cmake
index 010696a610..c5f9b7f341 100644
--- a/cmake/onnxruntime.cmake
+++ b/cmake/onnxruntime.cmake
@@ -83,6 +83,8 @@ if(onnxruntime_BUILD_SHARED_LIB)
       "${ONNXRUNTIME_ROOT}/core/dll/delay_load_hook.cc"
       "${ONNXRUNTIME_ROOT}/core/dll/onnxruntime.rc"
     )
+    # Workaround for https://github.com/conda-forge/onnxruntime-feedstock/pull/56#issuecomment-1586080419
+    set_target_properties(onnxruntime PROPERTIES OUTPUT_NAME onnxruntime_conda)
   elseif(onnxruntime_BUILD_APPLE_FRAMEWORK)
     # apple framework requires the header file be part of the library
     onnxruntime_add_shared_library(onnxruntime
